.github/workflows/build_main.yml

```diff
@@ -62,170 +67,224 @@ on:
+
+      # If a custom nuget feed is used, we need to add it when setting up dotnet
+      - if: ${{ steps.check_build_time_feed.outputs.defined == 'true' }}
+        name: Setup .NET SDK (with custom feed)
+        uses: actions/setup-dotnet@v4.3.1
+        with:
+          dotnet-version: ${{ inputs.dotnet_version }}
+          source-url: ${{ secrets.build_time_nuget_feed_url }}
+        env:
+          NUGET_AUTH_TOKEN: ${{ secrets.build_time_nuget_feed_api_key }}
+
+      ###############################
+      # SETUP ENVIRONMENT IF NEEDED #
+      ###############################
+      - if: ${{ inputs.env_var_name_1 != '' }}
+        name: Set env var 1
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_1 }}"
+          echo "${{ inputs.env_var_name_1 }}=${{ secrets.env_var_value_1 }}" >> $env:GITHUB_ENV
+
+      - if: ${{ inputs.env_var_name_2 != '' }}
+        name: Set env var 2
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_2 }}"
+          echo "${{ inputs.env_var_name_2 }}=${{ secrets.env_var_value_2 }}" >> $env:GITHUB_ENV
+
+      - if: ${{ inputs.env_var_name_3 != '' }}
+        name: Set env var 3
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_3 }}"
+          echo "${{ inputs.env_var_name_3 }}=${{ secrets.env_var_value_3 }}" >> $env:GITHUB_ENV
+
+      - if: ${{ inputs.env_var_name_4 != '' }}
+        name: Set env var 4
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_4 }}"
+          echo "${{ inputs.env_var_name_4 }}=${{ secrets.env_var_value_4 }}" >> $env:GITHUB_ENV
+
+      - if: ${{ inputs.env_var_name_5 != '' }}
+        name: Set env var 5
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_5 }}"
+          echo "${{ inputs.env_var_name_5 }}=${{ secrets.env_var_value_5 }}" >> $env:GITHUB_ENV
+
+      - if: ${{ inputs.env_var_name_6 != '' }}
+        name: Set env var 6
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_6 }}"
+          echo "${{ inputs.env_var_name_6 }}=${{ secrets.env_var_value_6 }}" >> $env:GITHUB_ENV
+
+      - if: ${{ inputs.env_var_name_7 != '' }}
+        name: Set env var 7
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_7 }}"
+          echo "${{ inputs.env_var_name_7 }}=${{ secrets.env_var_value_7 }}" >> $env:GITHUB_ENV
+
+      - if: ${{ inputs.env_var_name_8 != '' }}
+        name: Set env var 8
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_8 }}"
+          echo "${{ inputs.env_var_name_8 }}=${{ secrets.env_var_value_8 }}" >> $env:GITHUB_ENV
+
+      ###############################
+      # Install required tools      #
+      ###############################
+      - name: Enable global .NET tools
+        run: |
+          echo $env:HOME\.dotnet\tools >> $GITHUB_PATH
+
+      - name: Install Roslynator
+        run: |
+          dotnet tool install --global Roslynator.DotNet.Cli
```

Comment:
Let's lock in to some specific version for this to avoid breaking changes breaking our builds in the future.